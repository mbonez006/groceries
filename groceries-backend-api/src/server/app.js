import express from "express";
import bodyParser from "body-parser";
import {
  addItem,
  getAllGroceries,
  markItemAsPurchased,
} from "../api/groceries";

const app = express();

app.use(bodyParser.urlencoded({ extended: true }));
app.use(bodyParser.json());

app.get("/groceries", getAllGroceries);
app.post("/grocery/status", markItemAsPurchased);
app.post("/grocery/add", addItem);

module.exports = app;

export default app;
