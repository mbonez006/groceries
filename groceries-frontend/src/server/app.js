import express from "express";
import bodyParser from "body-parser";
import path from "path";
import { groceriesHandler, groceriesPostHandler } from "../pages/groceries";

const app = express();

app.use(bodyParser.urlencoded({ extended: true }));
app.set("views", path.join(__dirname, "../pages"));
app.set("view engine", "pug");
app.use("/app.js", express.static("dist/main.bundle.js"));
app.use("/style.css", express.static("dist/style.css"));
app.use("/logo.png", express.static("src/public/img/logo.png"));

app.get("/", groceriesHandler);
app.post("/", groceriesPostHandler, groceriesHandler);

module.exports = app;

export default app;
