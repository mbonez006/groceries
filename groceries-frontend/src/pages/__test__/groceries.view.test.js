import request from "supertest";
import cheerio from "cheerio";
import app from "../../server/app";

describe("Groceries view", () => {
  let response;
  let $;

  beforeAll(async () => {
    app.get("/mock-groceries-view", (req, res) => {
      res.locals.groceries = ["Milk", "Coffee"];
      res.render("groceries.pug");
    });

    response = await request(app).get("/mock-groceries-view");
    $ = cheerio.load(response.text);
  });

  it("should have expected browser tab title", () => {
    const title = $("title").text();
    expect(title).toBe("Groceries");
  });

  it("should have heading as expected", () => {
    const heading = $("h3").text();
    expect(heading).toBe("Groceries");
  });

  it("should have list as expected", () => {
    const listItem = $("td").length;
    expect(listItem).toBe(4);
  });

  it("should have add grocery heading", () => {
    const addHeading = $("#add-grocery").text().trim();
    expect(addHeading).toEqual("Add Grocery Item");
  });

  it("should have a snapshop of page", () => {
    const page = $.html();
    expect(page).toMatchSnapshot();
  });
});
