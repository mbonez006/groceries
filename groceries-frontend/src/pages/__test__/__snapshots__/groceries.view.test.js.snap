// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Groceries view should have a snapshop of page 1`] = `"<!DOCTYPE html><html><head><title>Groceries</title><meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1, shrink-to-fit=no\\"><link rel=\\"stylesheet\\" href=\\"/style.css\\"></head><body><div class=\\"container\\"><div class=\\"row\\"><div class=\\"col-md-12\\">          <div class=\\"pageHeader\\"><img class=\\"logo\\" src=\\"logo.png\\" width=\\"150\\" height=\\"50\\"><h3>Groceries</h3></div></div></div><div class=\\"row\\"> <div class=\\"col-md-9\\" style=\\"border-right: solid;\\"><table class=\\"table center\\"><thead><tr><th scope=\\"col\\">Item</th><th scope=\\"col\\">Action</th></tr></thead><tbody><tr><td><s></s></td><td>                    <button class=\\"btn btn-secondary\\" type=\\"button\\" disabled=\\"\\">Purchased</button></td></tr><tr><td><s></s></td><td>                    <button class=\\"btn btn-secondary\\" type=\\"button\\" disabled=\\"\\">Purchased</button></td></tr></tbody></table></div><div class=\\"col-md-3\\"> <h5 class=\\"text-center\\" id=\\"add-grocery\\" style=\\"padding-top: 5%;\\">Add Grocery Item</h5><form action=\\"/\\" method=\\"POST\\"><div class=\\"row\\"><div class=\\"col-md-12\\"><input class=\\"form-control\\" type=\\"text\\" name=\\"itemToAdd\\" placeholder=\\"item*\\" required=\\"true\\" style=\\"margin-bottom: 5%;\\"></div></div><div class=\\"row\\"><div class=\\"col-md-12\\"><button class=\\"btn btn-info btn-block waves-effect waves-light\\" type=\\"submit\\">Submit</button></div></div></form></div></div></div></body></html>"`;
